### PR TITLE
RPE-31: Create an initial version of Spring Boot application template

### PR DESCRIPTION
**Pull request by lgonczar**
It's the very basic version, created using spring initializr, with some minor changes. Dependencies, plugins, code quality tool configuration, etc. will come later.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.5.1-bin.zip
```
Comment (rkondratowicz) [inline gradle/wrapper/gradle-wrapper.properties]: the latest version is `4.1` I believe

File: src/main/java/uk/gov/hmcts/reform/demo/DemoApplication.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.hmcts.reform.demo;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+// TODO: change the name of this class to one matching your project, e.g. DraftStoreApplication
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/demo/DemoApplication.java]: isn't just `Application.java` fine for all projects?

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+repositories {
+	mavenCentral()
+}
+
+
+dependencies {
+	compile('org.springframework.boot:spring-boot-starter')
+	testCompile('org.springframework.boot:spring-boot-starter-test')
+}
```
Comment (rkondratowicz) [inline build.gradle]: use spaces instead of tabs

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,27 @@
+buildscript {
```
Comment (timja-kainos) [inline build.gradle]: buildscript is old and deprecated,
should be using new plugins mechanism.

example one:
```
plugins {
    id 'application'
    id 'checkstyle'
    id 'io.spring.dependency-management' version '1.0.3.RELEASE'
    id 'org.springframework.boot' version '1.5.6.RELEASE'
    id 'org.owasp.dependencycheck' version '1.4.5.1'
    id 'com.github.ben-manes.versions' version '0.15.0'
    id 'org.sonarqube' version '2.5'
    id 'jacoco'
}
```

File: build.gradle (hunk 3 above)
Comment (rkondratowicz) [reply build.gradle]: in all java classes below as well

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	dependencies {
+		classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
+	}
+}
+
+apply plugin: 'java'
+apply plugin: 'org.springframework.boot'
+
+version = '0.0.1-SNAPSHOT'
```
Comment (timja-kainos) [inline build.gradle]: SNAPSHOT is more a maven thing than gradle.

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+sourceCompatibility = 1.8
+
+repositories {
+	mavenCentral()
```
Comment (timja-kainos) [inline build.gradle]: jcenter is used here more than mavenCentral, any reason you've defaulted to mavenCentral?

File: build.gradle [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+
+dependencies {
+	compile('org.springframework.boot:spring-boot-starter')
```
Comment (timja-kainos) [inline build.gradle]: this format is easier to read imo:
```
compile group: 'org.springframework.boot', name: 'spring-boot-starter', version: <version if specified>
```

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (lgonczar) [reply gradle/wrapper/gradle-wrapper.properties]: I'l change that

File: src/main/java/uk/gov/hmcts/reform/demo/DemoApplication.java (hunk 2 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/demo/DemoApplication.java]: That was my first plan, but then I saw DraftStoreApplication in draft-store (don't have access to internal repositories).

File: build.gradle (hunk 3 above)
Comment (lgonczar) [reply build.gradle]: Sorry, autogenerated. Will change.

File: .gitignore [hunk 8]
```diff
@@ -1 +1,19 @@
+.gradle
```
Comment (timja-kainos) [inline .gitignore]: I would recommend also providing:
* editorconfig file
* Jenkinsfile
* travis ci file
* codecoverage example

Comment (lgonczar) [reply .gitignore]: All of these are going to be added as a part of the story, but as I mentioned in the description, this is the very basic one. I don't want to add everything in one pull request.

File: build.gradle (hunk 6 above)
Comment (lgonczar) [reply build.gradle]: This file was generated by spring initializr - that's how mavenCentral got here. My plan was to make it the initial version and refine, but will change this bit now.

File: build.gradle (hunk 7 above)
Comment (lgonczar) [reply build.gradle]: It is easier to read. Will change.

File: build.gradle (hunk 4 above)
Comment (lgonczar) [reply build.gradle]: Done. For now, just Spring-related plugins, but I'll add whatever is appropriate pull requests after this one.

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-bin.zip
```
Comment (timja-kainos) [inline gradle/wrapper/gradle-wrapper.properties]: I would recommend this version instead:
```
distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```

Gives IDE gradle sources for autocomplete

Comment (lgonczar) [reply gradle/wrapper/gradle-wrapper.properties]: Done

**Pull request merged**